src/NMonad/Core.hs

```diff
@@ -11,12 +28,45 @@ module NMonad.Core
   , fromTimeout
   , runN
 
+  -- Lenses
+
+  -- DBusNotification
+  , replacesId
+
+  -- TODO: Find a better way to export these with TemplateHaskell
```

Comment:
I agree and suport this `TODO`!

Reply:
Sadly it doesn't seem like hoisting of any kind is possible with Template Haskell. Have a read of [the original paper](https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.189.4479&rep=rep1&type=pdf) and if you spot anything I could use, let me know!